### PR TITLE
Feat: add balance sorter for smart contracts list(Issue 1172)

### PR DESCRIPTION
**Pull request by Naupio**
## What problem does this PR solve?
issue: #1172 
## Check List
- check smart contract ckb_balance
- check smart contract ckb_balance auto update
#### Test
- unit test
#### Task
 - none

### CODE REVIEW

**Comment by FrederLu** [comment]
When the cursor is brought, 500 will be reported.
```graphql
query {
	smart_contracts(
		input: {
			limit: 5
			sorter: [
				{ sort_type: ASC, sort_value: CKB_BALANCE }
# 				{ sort_type: ASC, sort_value: ID }
# 				{ sort_type: ASC, sort_value: NAME }
			]
			after: "g3QAAAAA"
		}
	) {
		entries {
			name
			account_id
			account {
				eth_address
			}
			ckb_balance
		}
		metadata {
			total_count
			after
			before
		}
	}
}
```

**Comment by Naupio** [comment]
> When the cursor is brought, 500 will be reported.
> 
> ```graphql
> query {
> 	smart_contracts(
> 		input: {
> 			limit: 5
> 			sorter: [
> 				{ sort_type: ASC, sort_value: CKB_BALANCE }
> # 				{ sort_type: ASC, sort_value: ID }
> # 				{ sort_type: ASC, sort_value: NAME }
> 			]
> 			after: "g3QAAAAA"
> 		}
> 	) {
> 		entries {
> 			name
> 			account_id
> 			account {
> 				eth_address
> 			}
> 			ckb_balance
> 		}
> 		metadata {
> 			total_count
> 			after
> 			before
> 		}
> 	}
> }
> ```

fixed https://github.com/Magickbase/godwoken_explorer/pull/1257/commits/269883af80bb05fb466bc3c00e3c71cc48b80c89

**Comment by Naupio** [comment]
pls use new cursor to testing
```
query {
  smart_contracts(
    input: {
      limit: 5
      sorter: [{ sort_type: ASC, sort_value: CKB_BALANCE }]
      after: "g3QAAAACZAACaWRhB2gCZAAKY2tiX3NvcnRlZGQAC2NrYl9iYWxhbmNldAAAAARkAApfX3N0cnVjdF9fZAAORWxpeGlyLkRlY2ltYWxkAARjb2VmYQBkAANleHBhAGQABHNpZ25hAQ=="
    }
  ) {
    entries {
      name
      account_id
      account {
        eth_address
      }
      ckb_balance
    }
    metadata {
      total_count
      after
      before
    }
  }
}
```

```json
{
  "data": {
    "smart_contracts": {
      "entries": [
        {
          "account": {
            "eth_address": "0xa1596a70a65a6022c6db54404da72e19d32ec1e0"
          },
          "account_id": 59924,
          "ckb_balance": "0",
          "name": "Token"
        },
        {
          "account": {
            "eth_address": "0x2f5dd7f0b8325bc119261f8ff6a7f67e27502f75"
          },
          "account_id": 60322,
          "ckb_balance": "0",
          "name": ""
        },
        {
          "account": {
            "eth_address": "0xfed57d672852c7ed93ce70f3989ef1f1ac7ce164"
          },
          "account_id": 60319,
          "ckb_balance": "0",
          "name": ""
        },
        {
          "account": {
            "eth_address": "0x204715dfa4fbb5601ff7a633b5d4230ea5565e1f"
          },
          "account_id": 60320,
          "ckb_balance": "0",
          "name": ""
        },
        {
          "account": {
            "eth_address": "0x835611ca4688e6f114caf614c22f7b18301df9c0"
          },
          "account_id": 60317,
          "ckb_balance": "0",
          "name": ""
        }
      ],
      "metadata": {
        "after": "g3QAAAACZAACaWRhDWgCZAAKY2tiX3NvcnRlZGQAC2NrYl9iYWxhbmNldAAAAARkAApfX3N0cnVjdF9fZAAORWxpeGlyLkRlY2ltYWxkAARjb2VmYQBkAANleHBhAGQABHNpZ25hAQ==",
        "before": "g3QAAAACZAACaWRhCGgCZAAKY2tiX3NvcnRlZGQAC2NrYl9iYWxhbmNldAAAAARkAApfX3N0cnVjdF9fZAAORWxpeGlyLkRlY2ltYWxkAARjb2VmYQBkAANleHBhAGQABHNpZ25hAQ==",
        "total_count": 10000
      }
    }
  }
}
```

**Comment by Naupio** [comment]
we had add contract_addresses field for input object
testnet-stg example:
```grapqhl
query {
  smart_contracts(
    input: {
      contract_addresses: ["0xe05d380839f32bc12fb690aa6fe26b00bd982613"]
    }
  ) {
    entries {
      name
      account_id
      account {
        eth_address
      }
      ckb_balance
    }
  }
}
```
```json
{
  "data": {
    "smart_contracts": {
      "entries": [
        {
          "account": {
            "eth_address": "0xe05d380839f32bc12fb690aa6fe26b00bd982613"
          },
          "account_id": 375,
          "ckb_balance": "79000069999999999999",
          "name": "ERC20"
        }
      ]
    }
  }
}
```

**Review by zmcNotafraid — CHANGES_REQUESTED** [review]

File: lib/godwoken_indexer/worker/update_smart_contract_ckb.ex [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      from(a in Account,
+        where: a.eth_address in ^addresses,
+        join: s in SmartContract,
+        on: s.account_id == a.id,
+        select: %{s: s, id: a.id, address: a.eth_address}
+      )
+
+    result = Repo.all(q)
+    addresses = result |> Enum.map(fn m -> m[:address] end)
+    addresses_balance = CurrentUDTBalance.get_ckb_balance(addresses)
```
[the remark this answers is not in the document]
Comment (Naupio) [reply lib/godwoken_indexer/worker/update_smart_contract_ckb.ex:162]: fix with this commit https://github.com/Magickbase/godwoken_explorer/pull/1257/commits/dcbd3aac0828c95b538bd5eaf27df5f255c04277

**Review by FrederLu — APPROVED** [review]

File: lib/godwoken_indexer/worker/update_smart_contract_ckb.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (Naupio) [reply lib/godwoken_indexer/worker/update_smart_contract_ckb.ex:160]: add more trigger point https://github.com/Magickbase/godwoken_explorer/pull/1257/commits/f2498c50fcbd94fef18b9f1e0e546f8f180aa7b0

**Review by zmcNotafraid — APPROVED** [review]

**Pull request merged**